app/downstream_payload.rb

```diff
@@ -52,27 +44,40 @@ def unpublishing
     edition.unpublishing
   end
 
-  def content_payload
-    Presenters::EditionPresenter.new(
-      edition, draft: draft
-    ).for_content_store(payload_version)
+  def content_presenter
+    Presenters::EditionPresenter.new(edition, draft: draft)
+  end
+
+  def redirect_presenter
+    RedirectPresenter.from_edition(edition)
+  end
+
+  def gone_presenter
+    GonePresenter.from_edition(edition)
   end
 
-  def redirect_payload
-    RedirectPresenter.present(
-      base_path: base_path,
-      publishing_app: edition.publishing_app,
-      public_updated_at: unpublishing.created_at,
-      redirects: unpublishing.redirects,
-    ).merge(payload_version: payload_version)
+  def vanish_presenter
+    VanishPresenter.from_edition(edition)
   end
```

Comment:
Just noticed something here that I think may have slipped in a while ago (may be more appropriate to do in a separate PR if no other changes are needed here)

In [0b736c9](https://github.com/alphagov/publishing-api/commit/0b736c9717e6015363630b6cf131754fa5ff243e) we dropped memoizing the presenter, which (if I understand correctly) probably means we're doing link expansion twice on a publish unnecessarily. I think some memoizing of the presenters would stop this.